### PR TITLE
[svt] Fix download of wrong episode and update test

### LINKED ISSUE #5595: Dowloading file from svtplay.se sometimes results in the wrong file being downloaded
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Sweden

### Provide a description that is worded well enough to be understood

Trying to download a video with URLs of format https://www.svtplay.se/video/${id}/${name} sometimes results in the wrong file being downloaded.

Examples of breaking URLs: 
- https://www.svtplay.se/video/K5wLrGb/deadline
- https://www.svtplay.se/video/j4zJ9Nn/bilranaren

For videos that don't require this verification the downloads seem to work fine.

Exampels of working URLs:
- https://www.svtplay.se/video/jLdDpwA/den-basta-sommaren
- https://www.svtplay.se/video/8YyBQXp/handerna

I believe the extractor for svtplay.se will need to be updated, but I don't have enough insight into the inner workings of this project to provide a solution.

 In the section below, attempting to download https://www.svtplay.se/video/K5wLrGb/deadline results in the file https://www.svtplay.se/video/KBM7ZXx/mid90s from being downloaded.

**Edit**: Edited this section to remove parts that I no longer believe to be relevant to this problem.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.svtplay.se/video/K5wLrGb/deadline']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (linux_exe)
[debug] Python 3.10.6 (CPython x86_64 64bit) - Linux-6.0.9-arch1-1-x86_64-with-glibc2.36 (OpenSSL 3.0.7 1 Nov 2022, glibc 2.36)
[debug] exe versions: ffmpeg N-62475-g130d19bf20-static (setts), ffprobe N-62475-g130d19bf20-static, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] Using fake IP 78.76.216.231 (SE) as X-Forwarded-For
[debug] [SVTPlay] Extracting URL: https://www.svtplay.se/video/K5wLrGb/deadline
[SVTPlay] K5wLrGb: Downloading webpage
[SVTPlay] KM3WEXY: Downloading JSON metadata
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading m3u8 information
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading m3u8 information
[SVTPlay] KM3WEXY: Downloading m3u8 information
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading MPD manifest
[SVTPlay] KM3WEXY: Downloading m3u8 information
[SVTPlay] KM3WEXY: Downloading m3u8 information
[SVTPlay] KM3WEXY: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] KM3WEXY: Downloading 1 format(s): dash-hbbtv-hevc-0+dash-full-15
[debug] Invoking dashsegments downloader on "https://svt-vod-9g.akamaized.net/d0/se/20221104/68d3e83a-6bf3-4c68-baa1-01e2516f46ce/dash-hbbtv-hevc.mpd"
[dashsegments] Total fragments: 1281
[download] Destination: Mid90s [KM3WEXY].fdash-hbbtv-hevc-0.mp4
[download] 100% of    1.77GiB in 00:01:51 at 16.28MiB/s
[debug] Invoking dashsegments downloader on "https://svt-vod-9g.akamaized.net/d0/se/20221104/68d3e83a-6bf3-4c68-baa1-01e2516f46ce/dash-full.mpd"
[dashsegments] Total fragments: 1281
[download] Destination: Mid90s [KM3WEXY].fdash-full-15.m4a
[download] 100% of  113.54MiB in 00:00:58 at 1.95MiB/s
[Merger] Merging formats into "Mid90s [KM3WEXY].mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:Mid90s [KM3WEXY].fdash-hbbtv-hevc-0.mp4' -i 'file:Mid90s [KM3WEXY].fdash-full-15.m4a' -c copy -map 0:v:0 -map 1:a:0 -movflags +faststart 'file:Mid90s [KM3WEXY].temp.mp4'
Deleting original file Mid90s [KM3WEXY].fdash-full-15.m4a (pass -k to keep)
Deleting original file Mid90s [KM3WEXY].fdash-hbbtv-hevc-0.mp4 (pass -k to keep)
```

### PR DESCRIPTION
**Pull request by wader**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Based on patch by @dirkf in comment https://github.com/yt-dlp/yt-dlp/issues/5595#issuecomment-1427036023 Removes some REs that now accidentially finds id in a newly introduced __NEXT_DATA__ JSON object.

Update SVTPlay test to a hopefully more stable URL (Rederiet episode 1, has no expire date)

Fixes #5595

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🛠️📝🌐

Improved SVTPlayIE extractor by fixing a bug, simplifying code, and adding features. `yt_dlp/extractor/svt.py` now handles more metadata and subtitles for SVTPlay videos.

> _Rise from the ashes of the broken test case_
> _Simplify the extraction of the video ID_
> _Unleash the power of the metadata and subtitles_
> _SVTPlayIE is the extractor of the night_

### Walkthrough
*  Simplify and update regex patterns for video ID extraction in SVTPlayIE extractor ([link](https://github.com/yt-dlp/yt-dlp/pull/7789/files?diff=unified&w=0#diff-85262bc9d32449791ee19b99cb38e9981aa1b8e9cdea0387570051bc18e1de2cL253-R260))
* Update test case for SVTPlayIE extractor with a new video URL and ID, and add more metadata fields and subtitles ([link](https://github.com/yt-dlp/yt-dlp/pull/7789/files?diff=unified&w=0#diff-85262bc9d32449791ee19b99cb38e9981aa1b8e9cdea0387570051bc18e1de2cL148-R167))

</details>

### CODE REVIEW

**Comment by bashonly** [comment]
Add a new test instead of replacing an old one. If the old test no longer works, add a `skip` key to the test dict (the value should be the reason why it's being skipped)

Why does it have the same md5 as the old test?

**Comment by wader** [comment]
> Add a new test instead of replacing an old one. If the old test no longer works, add a `skip` key to the test dict (the value should be the reason why it's being skipped)

Ok, add skip reason. No sure what is going with md5 hmm

```
$ python3 test/test_download.py TestDownload.test_SVTPlay
Skipping SVTPlay: Episode is no longer available
s
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK (skipped=1)

$ python3 test/test_download.py TestDownload.test_SVTPlay_1
[debug] Loaded 1864 extractors
[debug] Using fake IP 78.69.195.245 (SE) as X-Forwarded-For
[SVTPlay] Extracting URL: https://www.svtplay.se/video/emBxBQj
[SVTPlay] emBxBQj: Downloading webpage
[SVTPlay] eyBd9aj: Downloading JSON metadata
[SVTPlay] eyBd9aj: Downloading MPD manifest
[SVTPlay] eyBd9aj: Downloading m3u8 information
[SVTPlay] eyBd9aj: Downloading MPD manifest
[SVTPlay] eyBd9aj: Downloading m3u8 information
[SVTPlay] eyBd9aj: Downloading MPD manifest
[SVTPlay] eyBd9aj: Downloading m3u8 information
[SVTPlay] eyBd9aj: Downloading MPD manifest
[SVTPlay] eyBd9aj: Downloading m3u8 information
[SVTPlay] eyBd9aj: Downloading m3u8 information
[SVTPlay] eyBd9aj: Downloading MPD manifest
[SVTPlay] eyBd9aj: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] eyBd9aj: Downloading 1 format(s): dash-hbbtv-avc-0
[info] Writing video metadata as JSON to: test_SVTPlay_1_eyBd9aj.info.json
.
----------------------------------------------------------------------
Ran 1 test in 2.388s

OK
```

**Comment by bashonly** [comment]
Don't worry about md5, I edited my comment after realizing there was a `skip_download` param in the test

In the future, please avoid force-pushing. All commits will squashed before merge

Patch looks good

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/svt.py [hunk 1]
```diff
@@ -168,6 +169,36 @@ class SVTPlayIE(SVTPlayBaseIE):
...
+            'series': 'Rederiet',
+            'subtitles': {
+                'sv': [{
+                    'ext': 'vtt',
+                }, {
+                    'ext': 'vtt',
+                }, {
+                    'ext': 'vtt',
+                }]
+            },
```
Comment (pukkandan) [inline yt_dlp/extractor/svt.py:194]:
```suggestion
                'sv': 'count:3'
            },
```
is good enough

File: yt_dlp/extractor/svt.py [hunk 2]
```diff
@@ -168,6 +169,36 @@ class SVTPlayIE(SVTPlayBaseIE):
...
+        },
+        'params': {
+            # skip for now due to download test asserts that segment is > 10000 bytes and svt uses
+            # init segments that are smaller
+            # AssertionError: Expected test_SVTPlay_jNwpV9P.mp4 to be at least 9.77KiB, but it's only 864.00B
+            'skip_download': True,
```
Comment (pukkandan) [inline yt_dlp/extractor/svt.py:200]:
```suggestion
            'skip_download': 'm3u8',
```

This is a known issue affecting many IEs. Extra comment isnt needed

File: yt_dlp/extractor/svt.py [hunk 3]
```diff
@@ -250,12 +281,8 @@ def _real_extract(self, url):
         if not svt_id:
             svt_id = self._search_regex(
                 (r'<video[^>]+data-video-id=["\']([\da-zA-Z-]+)',
-                 r'<[^>]+\bdata-rt=["\']top-area-play-button["\'][^>]+\bhref=["\'][^"\']*video/%s/[^"\']*\b(?:modalId|id)=([\da-zA-Z-]+)' % re.escape(video_id),
-                 r'["\']videoSvtId["\']\s*:\s*["\']([\da-zA-Z-]+)',
-                 r'["\']videoSvtId\\?["\']\s*:\s*\\?["\']([\da-zA-Z-]+)',
-                 r'"content"\s*:\s*{.*?"id"\s*:\s*"([\da-zA-Z-]+)"',
-                 r'["\']svtId["\']\s*:\s*["\']([\da-zA-Z-]+)',
-                 r'["\']svtId\\?["\']\s*:\s*\\?["\']([\da-zA-Z-]+)'),
+                 r'<[^>]+\bdata-rt=["\']top-area-play-button["\'][^>]+\bhref=["\'][^"\']*video/[\da-zA-Z-]+/[^"\']*\b(?:modalId|id)=([\da-zA-Z-]+)',
```
Comment (pukkandan) [inline yt_dlp/extractor/svt.py:284]:
```suggestion
                 r'<[^>]+\bdata-rt=["\']top-area-play-button["\'][^>]+\bhref=["\'][^"\']*video/[\w-]+/[^"\']*\b(?:modalId|id)=([\w-]+)',
```

**Review by pukkandan — APPROVED** [review]

**Comment by wader** [comment]
Thanks. Successfully ran tests and tried a bit with some random svtplay URLs, then i noticed that there are URLs that work with the old version but not with these changes like https://www.svtplay.se/klipp/e9dBZ9R/det-kan-handa-efter-prigozjins-formodade-dod

**Comment by wader** [comment]
Re-adding `r'["\']videoSvtId\\?["\']\s*:\s*\\?["\']([\da-zA-Z-]+)',` seems to fix https://www.svtplay.se/klipp/e9dBZ9R/det-kan-handa-efter-prigozjins-formodade-dod

**Comment by bashonly** [comment]
> Re-adding `r'["\']videoSvtId\\?["\']\s*:\s*\\?["\']([\da-zA-Z-]+)'`, seems to fix...

Wasn't that one of the problematic patterns that was matching the wrong video ID? If we re-add more back, what is the point of this PR? Should more research be done into which patterns yield the wrong IDs?

**Comment by bashonly** [comment]
@wader Test my changes and let me know

**Comment by wader** [comment]
Tried a few old and new ones, all seem to work 👍

https://www.svtplay.se/klipp/e9dBZ9R/det-kan-handa-efter-prigozjins-formodade-dod - works
https://www.svtplay.se/video/emBxBQj - works
https://www.svtplay.se/video/jz2rYz7/anders-hansen-moter/james-fallon?info=visa - works
https://www.svtplay.se/video/e2ov9zm/aktuellt/ikvall-21-00 - works
https://www.svtplay.se/klipp/jb3by35/spadbarnsmordaren?info=visa - works

File: yt_dlp/extractor/svt.py [hunk 4]
```diff
@@ -247,15 +271,16 @@ def _real_extract(self, url):
                 data, lambda x: x['statistics']['dataLake']['content']['id'],
                 compat_str)
 
+        if not svt_id:
+            nextjs_data = self._search_nextjs_data(webpage, video_id, fatal=False)
```
Comment (wader) [inline yt_dlp/extractor/svt.py:275]: TIL huh that is a handy function

**Comment by wader** [comment]
Retried, all good 👍

**Comment by wader** [comment]
Let me know if i should help testing etc

**Pull request merged**